runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java

```diff
@@ -250,7 +249,12 @@ State waitUntilFinish(
       }
 
       if (!hasError) {
+        // Reset the backoff, and update the new cumulative sleep time to be the remaining
+        // part of the total input sleep duration.
         backoff.reset();
+        long nanosConsumed = nanoClock.nanoTime() - startNanos;
+        Duration consumed = Duration.millis((nanosConsumed + 999999) / 1000000);
+        backoff = backoff.withMaxCumulativeBackoff(duration.minus(consumed));
```

Comment:
The value tracked by the BackOff is a different number than the code here wants. The code here wants wall-clock consumed.